File: Data-Structure/Nodejs/Top Interview Questions/String/Easy/Reverse String.js
{
    //Two Pointers
    // T: O(N)
    // S: O(1)
    var reverseString = function (s) {
        //The input string is given as an array of characters s.
        if (s.length < 2) return s;
        const swap = (arr, i, j) => [arr[i], arr[j]] = [arr[j], arr[i]];
        let left = 0;
        let right = s.length - 1;
        while (left < right) {
            swap(s, left, right);
            left++;
            right--;
        }
        return s;
    }
}

File: Data-Structure/Alg-Study-Plan/双指针 Two Pointers/左右指针/00344. Reverse String/reverseString.java
public class Solution {
    void reverseString(char[] arr) {
        int left = 0;
        int right = arr.length - 1;
        while (left < right) {
            // 交换 arr[left] 和 arr[right]
            char temp = arr[left];
            arr[left] = arr[right];
            arr[right] = temp;
            left++;
            right--;
        }
    }
}

    
